### ISSUE TITLE
Performance issue relating to build queue

### ISSUE THREAD

**Issue by nilederg** [Anuken/Mindustry#6535]
**Platform**: *Windows*

**Build**: *Steam 126.3*

**Issue**: *Massive performance drop when there is a large amount in the build queue and a large amount loaded as a schematic. This does not happen when that amount is in the queue alone, or when it is loaded as a schem alone, this issue must be an interaction between the two.*

**Steps to reproduce**: 
[2022-01-26 13-25-09.zip](https://github.com/Anuken/Mindustry/files/7945721/2022-01-26.13-25-09.zip)
Apologies for the low video framerate, it does not represent the actual framerate in game. Refer to the FPS counter in-game.

**Link(s) to mod(s) used**: *None applicable*

**Save file**: 
[101.zip](https://github.com/Anuken/Mindustry/files/7945732/101.zip)

---

*Place an X (no spaces) between the brackets to confirm that you have read the line below.*  
- [X ] **I have updated to the latest release (https://github.com/Anuken/Mindustry/releases) to make sure my issue has not been fixed.**
- [X ] **I have searched the closed and open issues to make sure that this problem has not already been reported.**

**Comment by nilederg** [comment]
not a potato pc issue btw

**Comment by Anuken** [comment]
I am aware of this; it is not a bug. It's the reason why schematic sizes are capped.

**Comment by LeoSko** [comment]
> It's the reason why schematic sizes are capped.

How does this relate to the issue and how is that not a bug? It is fairly simple to reproduce while using no schematics at all and still experience this performance degradation. Holding an F-selected single block + covering the map in plans still produces same slowness.

**Comment by Anuken** [comment]
> how is that not a bug?

Because it's both expected (n^2 comparison with player queue to check for overlap) and performance-related. "I selected a massive area of 1x1 identical blocks and my game lags" is not a bug for the same reason "I spawned a thousand units in my map and now my game lags" is not a bug.

Selecting a 32x32 area of *single blocks* is extremely unusual in normal play; there is no valid reason to do so unless you are making sorter art (*not really a use case I support*) or intentionally trying to lag the game by covering everything in plans.

I'm not saying that this can't/shouldn't be improved, but it's not game-breaking and it really doesn't fit my definition of a "bug".
